### PR TITLE
Implemented the JWT checker middleware

### LINKED ISSUE #32: Add authorization middleware to endpoints
Add a middleware that checks the JWT sent with the request (if there is no JWT fail the request immediately) using the Bumbal API ( `/authenticate/check-token` is their endpoint). Have a look at https://go-chi.io/#/pages/middleware and maybe https://go-chi.io/#/pages/routing . 

Take your time and write clean code. Write tests.

### PR DESCRIPTION
**Pull request by VictorVader**

### CODE REVIEW

**Review by Patricijia — DISMISSED** [review]

File: backend/cmd/web/routes.go [hunk 1]
```diff
@@ -11,6 +12,37 @@ func hello(w http.ResponseWriter, r *http.Request) {
...
+//				no JWT, request fails immediately
+//	@param next
+//	@return http.Handler
+func JwtChecker(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		url := "https://sep202302.bumbal.eu/api/v2/authenticate/check-token"
+		jwToken := r.Header.Get("Authorization") // -> "Bearer <TOKEN>"
+
+		// query BUMBAL /check-token -> GET req with TOKEN in body
+		req, err := http.NewRequest(http.MethodGet, url, nil)
```
Comment (NeaGogu) [inline backend/cmd/web/routes.go:26]: check the error

File: backend/cmd/web/routes.go [hunk 2]
```diff
@@ -11,6 +12,37 @@ func hello(w http.ResponseWriter, r *http.Request) {
...
+		req.Header.Add("Authorization", jwToken)
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if resp.StatusCode != http.StatusOK {
+			fmt.Println(resp.StatusCode)
```
Comment (NeaGogu) [inline backend/cmd/web/routes.go]: remove the println (always remove them when you've tested your code and don't need them anymore). They pollute the logs

**Review by NeaGogu — CHANGES_REQUESTED** [review]

File: backend/cmd/web/routes.go (hunk 1 above)
Comment (VictorVader) [reply backend/cmd/web/routes.go:26]: Done

File: backend/cmd/web/routes.go (hunk 2 above)
Comment (VictorVader) [reply backend/cmd/web/routes.go]: Sir yes sir

**Review by NeaGogu — APPROVED** [review]
lgtm

**Review by Patricijia — APPROVED** [review]

**Pull request merged**